pkg/ring/replication_set.go

```diff
@@ -83,14 +83,37 @@ func (r ReplicationSet) Do(ctx context.Context, delay time.Duration, f func(cont
 					forceStart <- struct{}{}
 				}
 			} else {
-				results = append(results, res.res)
+				if _, ok := resultsPerZone[res.instance.Zone]; !ok {
+					resultsPerZone[res.instance.Zone] = make([]interface{}, 0, len(r.Instances))
+				}
+				resultsPerZone[res.instance.Zone] = append(resultsPerZone[res.instance.Zone], res.res)
```

Comment:
Is it possible to have 2 new methods on the tracker interface to hide this logic?

* tracker.RecordResult(instance, result)
* GetQuerumResults
  * That for sone aware will return only the zonal results?

Reply:
Updated. I combine `RecordResult` to the existing `done` method.
`GetResults` is a general method to return results. Quorum or not is controlled via the flag